clients/service/src/lib.rs

```diff
@@ -170,17 +173,38 @@ impl<Config: Clone + Send + 'static, F: Fn()> ConnectionManager<Config, F> {
 	}
 }
 
-pub async fn wait_or_shutdown<F, E>(shutdown_tx: ShutdownSender, future2: F) -> Result<(), E>
+pub async fn wait_or_shutdown<F, E>(
+	shutdown_tx: ShutdownSender,
+	future2: F,
+	// a consumer that receives a precheck signal to start a task.
+	precheck_signal: Option<tokio::sync::broadcast::Receiver<()>>,
+) -> Result<(), E>
 where
 	F: Future<Output = Result<(), E>>,
 {
+	if let Some(mut precheck_signal) = precheck_signal {
+		loop {
+			match precheck_signal.try_recv() {
+				// Received a signal to start the task
+				Ok(_) => break,
+				Err(TryRecvError::Empty) =>
+					tracing::trace!("wait_or_shutdown precheck signal: waiting..."),
```

Comment:
I think this is problematic because we are again creating tasks that never yield, potentially causing issues for the scheduling of other tasks. 
```suggestion
				Err(TryRecvError::Empty) => {
					tracing::trace!("wait_or_shutdown precheck signal: waiting..."),
					sleep(Duration::from_millis(100)).await; // Yield to the runtime
				}
```

Reply:
Or we can use `precheck_signal.recv().await`? this one is [async](https://docs.rs/tokio/latest/tokio/sync/broadcast/struct.Receiver.html#method.recv).

Reply:
I think if the receiver `precheck_signal` has already consumed the message then next time calling `recv().await`, it will yield until a new one arrives. Or do you think there will always be a message in the broadcast channel? 

This implementation is equivalent though. But with `try_recv` we do need to yield that's true.

Reply:
>  do you think there will always be a message in the broadcast channel?

It will only send 1 `()` message. See lines 335-337 in clients/vault/src/requests/execution.rs.

[The first try was to use `recv().await`](https://github.com/pendulum-chain/spacewalk/pull/545#discussion_r1714911315) but the possibility of it waiting forever is there. 

However.. we also cannot move forward with the task anyway, if the precheck (or the `execute_open_requests`) is not finished. 
I also would not like to wrap the `recv()` in a `timeout()` because open requests might really take longer than the timeout we set. Also, a `timeout` is another thread; That's 8 tasks requiring prechecks, meaning 8 additional threads. (See [Marcel's comment](https://github.com/pendulum-chain/spacewalk/pull/545#pullrequestreview-2360773756) for the image of these tasks: Issue Cancel Listener, Issue Request Listener, etc.)

Reply:
Readability, reverting to `recv().await` would be better. I would add that back, but we have to risk it awaiting. 
I think `execute_open_requests()` is reliable enough to be able to send the message fast.